napalm_nxos/nxos.py

```diff
@@ -241,6 +241,46 @@ def get_lldp_neighbors(self):
 
         return results
 
+
+    def get_bgp_neighbors(self):
+        cmd = 'show bgp sessions vrf all'
+        vrf_list = self._get_command_table(cmd, 'TABLE_vrf', 'ROW_vrf')
+
+        results = {}
+        for vrf_dict in vrf_list:
+            result_vrf_dict = {}
+            result_vrf_dict['router_id'] = unicode(vrf_dict['router-id'])
+            result_vrf_dict['peers'] = {}
+
+            neighbors_list = vrf_dict.get('TABLE_neighbor', {}).get('ROW_neighbor', [])
```

Comment:
It may be easier for us if you review the test case sample data, eg `test/unit/nxos/mock_data/show_bgp_sessions_single_neighbor.json` - I don't have access to equipment (sorry!) and have been making complete guesses as to what the data would be.  If I can get test cases, I'll make the code work.  Can I trouble you for that?  If you don't have time, np, I hope to get access to a lab soon.